### ISSUE TITLE
documentation request: how to use credentials + commit info

### ISSUE THREAD

**Issue by sdmoralesma** [jenkinsci/stashnotifier-plugin#120]
Currently the documentation to use the plugin with jenkins' pipeline is not clear how to inject the credentials in the `step()`, also, is not clear how should commit's hash be provided.

Without credentials you will receive an error like: 
`Failed to notify Stash for commit 26c235e2a6db947e0202446ac532350c5d24af0d ({"errors":[{"context":null,"message":"You are not permitted to access this resource","exceptionName":"com.atlassian.stash.exception.AuthorisationException"}]})`

**Comment by scaytrase** [comment]
as far as i understand, you should provide` `credentialsId` configuration for the step. But yes, full example for pipeline would be very useful

```
step([$class: 'StashNotifier', credentialsId: 'some-long-uuid-from-jenkins'])
```

**Comment by sdmoralesma** [comment]
Yes, people could even use the `pipeline syntax` that comes in jenkins, however the documentation here would be valuable. I add here an example as guideline:

```
step([$class: 'StashNotifier', 
              commitSha1: '$commit',  // jenkins parameter that will have commit's hash
              credentialsId: '000000-11bb-1234-1234-asdfqwer123',
              disableInprogressNotification: false, 
              ignoreUnverifiedSSLPeer: true, 
              includeBuildNumberInKey: false, 
              prependParentProjectKey: false, 
              projectKey: '', 
              stashServerBaseUrl: 'https://mystash.mycompany.com'])
```

**Comment by scaytrase** [comment]
you can propose this as README patch (additional to the short pipeline example), would be great

**Comment by sdmoralesma** [comment]
ok let me test the upgrade to v1.11 and if everything goes fine I will do it. :)

**Comment by scaytrase** [comment]
Closed with #121
